hugolib/pageSort.go

```diff
@@ -275,3 +276,20 @@ func (p Pages) Reverse() Pages {
 
 	return pages
 }
+
+func (p Pages) ByParam(paramsKeyStr string) Pages {
+	key := "pageSort.ByParam." + paramsKeyStr
+
+	paramsKeyComparator := func(p1, p2 *Page) bool {
+		v1, _ := p1.Param(paramsKeyStr)
```

Comment:
You understand that `Param` will look in the site config if param is not present in the page? I thought a little about this, and I think it makes sense.